### DISCUSSION TITLE
Question: PHP port feasibility

### DISCUSSION THREAD

**Discussion by Stamenov** [nipunsadvilkar/pySBD#110] [Q&A] [answered]
Hi there,

thank you for your efforts into porting pragmatic_segmenter to python, I have been using it successfully for some time now.

I am wondering, based on your experience, how difficult it would be to port it to PHP as well and what were some hurdles porting ruby to python?

Thank in advance,
Best

**Comment by nipunsadvilkar** [comment, marked as answer]
Hi @Stamenov 

Thanks for the appreciation.

> how difficult it would be to port it to PHP as well and what were some hurdles porting ruby to python?
It's very subjective question actually and depends on everyones skill set.

Best way I can recommend is to opt for just one language at first and use python debugger to see how your input text goes through different transformations to get clean sentence. As you understand each steps, write a code for that specific transformation in PHP. Repeat this cycle many times, initially it will be difficult to wrap your head around but if you persist, you'll be surprised to see how our own brain works 😅 bcz as you progress you'll find it quicker to translate many transformation. Hope this helps!
